From d5f4e30a0815ae900c43604465e7087236cd4e50 Mon Sep 17 00:00:00 2001
From: Tal Yacobi <tyacobi@habana.ai>
Date: Tue, 13 May 2025 11:24:02 +0300
Subject: [PATCH] [SW-221571] preboot: init deferred devices

Some devices are not supported in pcie card type
(e.g., in HL-338).

Check which of these devices are supported
and init them on runtime.

Change-Id: I6467533ec17d4f309b9fe00e539c7601021192d6
Signed-off-by: Tal Yacobi <tyacobi@habana.ai>
---

diff --git a/app/io/preboot.c b/app/io/preboot.c
index 4f0fb94..acfbf42 100644
--- a/app/io/preboot.c
+++ b/app/io/preboot.c
@@ -673,10 +673,21 @@
 	}
 }
 
+void hl_device_drivers_deferred_init(void)
+{
+	DEVICE_INIT_IF_EXIST(xdpe, XDPE132G5H_DEV);
+	DEVICE_INIT_IF_EXIST(xdpe, XDPE132G5H_0_DEV);
+	DEVICE_INIT_IF_EXIST(xdpe, XDPE132G5H_1_DEV);
+	DEVICE_INIT_IF_EXIST(xdpe, XDPE132G5H_2_DEV);
+	DEVICE_INIT_IF_EXIST(xdpp1100, XDPP1100_DEV);
+}
+
 void main(void)
 {
 	int key;
 
+	hl_device_drivers_deferred_init();
+
 	preboot_device_init();
 	preboot_plat_early_init();
 	watchdog_disable();
